wikia-robots-txt.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+if ( !defined( 'MW_NO_SETUP' ) ) {
+	define( 'MW_NO_SETUP', 1 );
+}
+
+require_once( __DIR__ . '/includes/WebStart.php' );
+require_once( __DIR__ . '/includes/Setup.php' );
+
+$allowRobots = ( $wgWikiaEnvironment === WIKIA_ENV_PROD || $wgRequest->getBool( 'forcerobots' ) );
+
+$robots = new RobotsTxt();
+
+if ( !$allowRobots ) {
+	// No crawling preview, verify, sandboxes, showcase, etc
+	$robots->disallowPath( '/' );
+} else {
+	// Sitemap
+	$robots->setSitemap( sprintf( 'http://%s/sitemap-index.xml', $_SERVER['SERVER_NAME'] ) );
+
+	// Special pages
+	$robots->disallowSpecialPages();
+
+	$robots->allowSpecialPage( 'Allpages' );
+	$robots->allowSpecialPage( 'CreateNewWiki' );
+	$robots->allowSpecialPage( 'Forum' );
+	$robots->allowSpecialPage( 'Sitemap' );
+	$robots->allowSpecialPage( 'Videos' );
+
+	// Params
+	$robots->disallowParam( 'action' );
+	$robots->disallowParam( 'feed' );
+	$robots->disallowParam( 'printable' );
+	$robots->disallowParam( 'useskin' );
+	$robots->disallowParam( 'uselang' );
+
+	// Nasty robots
+	$robots->blockRobot( 'IsraBot' );
```

Comment:
Might be nice to have these values as an array and loop through them to allow, disallow, block, etc. Would make this file more DRY.